src/uct/ib/rc/accel/rc_mlx5_ep.c

```diff
@@ -621,20 +621,75 @@ ucs_status_t uct_rc_mlx5_ep_fc_ctrl(uct_ep_t *tl_ep, unsigned op,
     return UCS_OK;
 }
 
+static void *uct_rc_mlx5_ep_pack_ece(uct_rc_mlx5_iface_common_t *iface,
+                                    uct_rc_mlx5_ep_address_t *rc_addr,
+                                    void *ptr, uint32_t ece, unsigned flag,
+                                    unsigned sr_flag)
+{
+    uct_ib_mlx5_ece_fields_t ece_fields;
+
+    if ((iface->config.flags & UCT_RC_MLX5_IFACE_FORCE_ECE) ||
+        (uct_ib_mlx5_decode_ece(ece, &ece_fields)) != UCS_OK) {
```

Comment:
compress_2ece(ece1, ece2, &addr_ece) -> 
c_ece.flags (up to ~4 flags)
c_ece.data_size (bytes)
c_ece.data[4]

rc_addr->flags |= unpacked_ece.flags
memcpy(rc_addr+1, c_ece.data, c_ece.data_size)